Realm/RLMArray.mm

```diff
@@ -168,6 +237,9 @@ - (RLMResults *)objectsWhere:(NSString *)predicateFormat args:(va_list)args
 }
 
 - (id)valueForKey:(NSString *)key {
+    if (!_backingArray) {
+        return @[];
```

Comment:
The change here is just that `_backingArray` is created lazily for standalone arrays (for reasons I don't remember; that may end up getting changed back if there's no longer a good reason for it), so a `nil` backing array needs to behave like an empty one.